pwa-shell/src/ui/common/SwipeContentNavigator.js

```diff
@@ -0,0 +1,92 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Gesture } from 'react-with-gesture';
+import { Redirect } from 'react-router-dom';
+
+class SwipeContentNavigator extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasSiblingContent: false,
+      xMin: 0,
+      xMax: 0,
+      openURL: '',
+    };
+    this.handleCursorMoving = this.handleCursorMoving.bind(this);
+  }
+
+  checkContentSiblings() {
+    const {
+      nextURL,
+      previousURL,
+    } = this.props;
+
+    let w = Math.max(document.documentElement.clientWidth, window.innerWidth || 0) / 2;
+    const hasNext = nextURL !== '';
+    const hasPrev = previousURL !== '';
+    this.setState({
+      hasSiblingContent: hasPrev || hasNext || false,
```

Comment:
is that `|| false` needed?

Reply:
this is funny i feel ashamed. lol i didn't know I placed that